.github/workflows/changelog-existence.yml

```diff
@@ -0,0 +1,106 @@
+# **what?**
+# Checks that a file has been committed under the /.changes directory
+# as a new CHANGELOG entry.  Cannot check for a specific filename as
+# it is dynamically generated by change type and timestamp.
+#
+
+# **why?**
+# Ensure code change gets reflected in the CHANGELOG.
+
+# **when?**
+# This will run when called in a workflow.
+
+# Example Usage including required permissions
+
+# permissions:
+#   contents: read
+#   pull-requests: write
+#
+# name: Check Changelog Entry
+# jobs:
+#   changelog:
+#     uses: dbt-labs/actions/.github/workflows/changelog-check.yml@main
+#    with:
+#      changelog_comment: 'Thank you for your pull request! We could not find a changelog entry for this change. For details on how to document a change, see [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-changelog-entry).'
```

Comment:
So this requires installing yet another program. This increases the learning curve for new contributors. There should at least be a setting to make this optional and not fail the pipeline.

Are there alternatives? Github can automatically generate a change log when creating a new release: https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes Can you see how it compares to changie?